### PR TITLE
Reduce scatter perf sweep

### PR DESCRIPTION
**Pull request by caixunshiren**
### Ticket
#11856 

### Problem description
Expose num workers and num buffers in reduce scatter at op level and added perf sweep test

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/ccl/reduce_scatter/reduce_scatter_pybind.cpp [hunk 1]
```diff
@@ -48,7 +52,7 @@ void py_bind_reduce_scatter(pybind11::module& module) {
     detail::bind_reduce_scatter(
         module,
         ttnn::reduce_scatter,
-        R"doc(reduce_scatter(input_tensor: std::vector<ttnn.Tensor>, scatter_dim: int, math_op: ReduceType, *, num_links: int = 1, memory_config: Optional[ttnn.MemoryConfig] = None) -> std::vector<ttnn.Tensor>
+        R"doc(reduce_scatter(input_tensor: std::vector<ttnn.Tensor>, scatter_dim: int, math_op: ReduceType, *, num_links: int = 1, memory_config: Optional[ttnn.MemoryConfig] = None, num_workers: int = 8, num_buffers_per_channel: int = 2) -> std::vector<ttnn.Tensor>
```
Comment (SeanNijjar) [inline ttnn/cpp/ttnn/operations/ccl/reduce_scatter/reduce_scatter_pybind.cpp:55]: Please fix the default arg values

**Review by SeanNijjar — APPROVED** [review]

**Comment by caixunshiren** [comment]
All post commits: https://github.com/tenstorrent/tt-metal/actions/runs/10892968502
t3k frequent: https://github.com/tenstorrent/tt-metal/actions/runs/10892987988
t3k nightly: https://github.com/tenstorrent/tt-metal/actions/runs/10892997897

File: ttnn/cpp/ttnn/operations/ccl/reduce_scatter/device/host/reduce_scatter_full_worker_grid.cpp [hunk 2]
```diff
@@ -750,13 +752,17 @@ operation::ProgramWithCallbacks reduce_scatter_with_workers(
         input_tensor_n_elems_per_slice / (tt::constants::TILE_WIDTH * tt::constants::TILE_HEIGHT);
 
     TT_ASSERT(input_tensor_num_units_per_tensor_slice > 0);
-    uint32_t max_num_workers = std::min<std::size_t>(8, input_tensor_num_units_per_tensor_slice);
+    uint32_t max_num_workers = std::min<std::size_t>(user_defined_num_workers.value_or(input_tensor_num_units_per_tensor_slice), input_tensor_num_units_per_tensor_slice);
```
Comment (TT-BrianLiu) [inline ttnn/cpp/ttnn/operations/ccl/reduce_scatter/device/host/reduce_scatter_full_worker_grid.cpp:755]: This behaviour will be different from before? Is this intended?

File: ttnn/cpp/ttnn/operations/ccl/reduce_scatter/reduce_scatter_pybind.cpp [hunk 3]
```diff
@@ -29,15 +29,19 @@ void bind_reduce_scatter(pybind11::module& module, const ccl_operation_t& operat
...
             py::arg("input_tensor"),
             py::arg("scatter_dim"),
             py::arg("math_op"),
             py::kw_only(),
             py::arg("num_links") = 1,
-            py::arg("memory_config") = std::nullopt});
+            py::arg("memory_config") = std::nullopt,
+            py::arg("num_workers") = std::nullopt,
+            py::arg("num_buffers_per_channel") = std::nullopt});
 }
```
Comment (TT-BrianLiu) [inline ttnn/cpp/ttnn/operations/ccl/reduce_scatter/reduce_scatter_pybind.cpp:45]: Are `num_links` and the two new args similar in nature? Should they all either have default values or be optional?

**Review by TT-BrianLiu — APPROVED** [review]
Minor comments. Please take a look

**Review by eyonland — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/ccl/reduce_scatter/reduce_scatter_pybind.cpp (hunk 3 above)
Comment (caixunshiren) [reply ttnn/cpp/ttnn/operations/ccl/reduce_scatter/reduce_scatter_pybind.cpp:45]: They should be optional. The default value is set in the kernel

File: ttnn/cpp/ttnn/operations/ccl/reduce_scatter/device/host/reduce_scatter_full_worker_grid.cpp (hunk 2 above)
Comment (caixunshiren) [reply ttnn/cpp/ttnn/operations/ccl/reduce_scatter/device/host/reduce_scatter_full_worker_grid.cpp:755]: good catch! this should be 
```
    uint32_t max_num_workers = std::min<std::size_t>(user_defined_num_workers.value_or(8), input_tensor_num_units_per_tensor_slice);
```
I will fix that

**Pull request merged**